velox/exec/GroupingSet.cpp

```diff
@@ -1081,6 +1094,48 @@ bool GroupingSet::mergeNext(
   VELOX_UNREACHABLE();
 }
 
+bool GroupingSet::mergeNextForDistinct(
+    int32_t maxOutputRows,
+    int32_t maxOutputBytes,
+    const RowVectorPtr& result) {
+  VELOX_CHECK(isDistinct());
+
+  bool nextKeyIsEqual{false};
+  bool newDistinct{true};
+  int32_t numOutputRows{0};
+  for (;;) {
+    const auto next = merge_->nextWithEquals();
+    auto* stream = next.first;
+    if (stream == nullptr) {
+      result->resize(numOutputRows);
```

Comment:
Do we need to check that numOutputRows > 0? Do we want to produce empty batches?

Reply:
We return a bool indicating whether the output is empty or not.